internal/pkg/daemon/bpfrecorder/bpf/recorder.bpf.c

```diff
@@ -257,6 +257,61 @@ int BPF_PROG(bprm_check_security, struct linux_binprm * bprm)
     return register_file_event(bprm->file, FLAG_SPAWN);
 }
 
+SEC("lsm/path_mkdir")
+int BPF_PROG(path_mkdir, struct path * dir, struct dentry * dentry, umode_t mode)
+{
+    // bpf_printk("path_mkdir");
+    u32 mntns = get_mntns();
+    if (!mntns)
+        return 0;
+
+    event_data_t * event;
+    event = bpf_ringbuf_reserve(&events, sizeof(event_data_t), 0);
+    if (!event) {
+        return 0;
+    }
+
+    int pathlen = bpf_d_path(dir, event->data, sizeof(event->data));
+    if (pathlen < 0) {
+        bpf_printk("register_file_event bpf_d_path failed: %i\n", pathlen);
+        bpf_ringbuf_discard(event, 0);
+        return 0;
+    }
+
+    // more checks than necessary, but only checking each offset
+    // individually makes the ebpf verifier happy.
+    if (
+        pathlen >= 1 &&
+        pathlen - 1 < sizeof(event->data) &&
+        pathlen + 0 < sizeof(event->data) &&
+        pathlen + 1 < sizeof(event->data) &&
+        pathlen + 2 < sizeof(event->data)
+    ) {
+        event->data[pathlen - 1] = '/';
+        event->data[pathlen - 0] = '*';
+        event->data[pathlen + 1] = '*';
+        event->data[pathlen + 2] = '\0';
+    } else {
+        bpf_printk("failed to fixup directory entry, not enough space: %s", event->data);
```

Comment:
This case is realistic only when the path exceeds the `PATH_MAX` size `4096` characters. Right?

I am wondering if it makes sense to overwrite the last last directory in the path with `**` in that case, but this is almost unrealistically to occur given the length of the path.

nit: maybe we should mention this in the error message.

Reply:
> This case is realistic only when the path exceeds the PATH_MAX size 4096 characters. Right?

Yes, correct - I'd say we look at workarounds iff we ever hit this. 
I've expanded the error message and added a comment. :)

Reply:
Perfect, thanks!